swing/src/net/sf/openrocket/gui/dialogs/ComponentAnalysisDialog.java

```diff
@@ -559,21 +548,28 @@ public void stateChanged(EventObject e) {
 			row.eachMass = cmEntry.eachMass;
 			row.cm = cmEntry.totalCM;
 
-//				forces = aeroData.get(c);
-//				if (forces == null) {
-//					continue;
-//				}
-//				if (forces.getCP() != null) {
-//					data[2] = forces;
-//				}
-
-//				if (!Double.isNaN(forces.getCD())) {
-//					dragData.add(forces);
-//				}
-//				if (c instanceof FinSet) {
-//					rollData.add(forces);
-//				}
-//			}
+			forces = aeroData.get(comp);
+			if (forces == null) {
+				System.err.println("Could not find aeroData entry for component: " + comp.getName());
+				row.cpx = 0.0;
+				row.cna = 0.0;
+				continue;
+			}
+
+//			System.err.println(String.format("        .CP = %s", forces.getCP()));
+			if (forces.getCP() != null) {
+				row.cpx = forces.getCP().x;
+				row.cna = forces.getCNa();
+			}
+
+//			System.err.println(String.format("        .CD = %s", forces.getCD()));
+			if (!Double.isNaN(forces.getCD())) {
+				dragData.add(forces);
+			}
+
+			if (comp instanceof FinSet) {
```

Comment:
Does this need to apply to TubeFinSets as well? They aren't part of the FinSet hierarchy...

edit: Thinking a bit about it, the TubeFinSets don't have a cant property so its likely this won't apply there. So I think the comment is moot, but leaving it here for others to agree or refute.

Reply:
Yea, I agree on both.   I'll add a comment for clarity.

Reply:
added a comment block right aftewards:
```
			// // We _would_ check this, except TubeFinSet doesn't implement cant angles... so they can't impart any roll torque
			// // If this is ever implemented, uncomment this block:
			// else if(comp instanceof TubeFinSet){
			// 	rollData.add(forces)
			// }
```